modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultGenerator.java

```diff
@@ -264,7 +264,8 @@ public Reader getTemplate(String name) {
                     operation.put("basePath", basePath);
                     operation.put("basePathWithoutHost", basePathWithoutHost);
                     operation.put("contextPath", contextPath);
-                    operation.put("baseName", tag);
+                   // operation.put("baseName", tag); FIXME ask @hiveship if this modification cause a problem
```

Comment:
I did it to generate a correct @Path annotation on the CXF interface but I think a better way to do would be:

-> Don't edit the DefaultGenerator (remove my modificaiton)
-> Update the mustache template file (JavaJaxRS/cxf/api.mustache)
```
package {{package}};

{{#imports}}import {{import}};
{{/imports}}

import javax.ws.rs.*;
import javax.ws.rs.core.Response;

@Path("{{baseName}}")
public interface {{classname}}  {
{{#operations}}
{{#operation}}
    @{{httpMethod}}
    {{#subresourceOperation}}@Path("{{path}}"){{/subresourceOperation}}
    {{#hasConsumes}}@Consumes({ {{#consumes}}"{{mediaType}}"{{#hasMore}}, {{/hasMore}}{{/consumes}} }){{/hasConsumes}}
    {{#hasProduces}}@Produces({ {{#produces}}"{{mediaType}}"{{#hasMore}}, {{/hasMore}}{{/produces}} }){{/hasProduces}}
    public Response {{nickname}}({{#allParams}}{{>queryParams}}{{>pathParams}}{{>headerParams}}{{>bodyParams}}{{>formParams}}{{#hasMore}},
    {{/hasMore}}{{/allParams}});
{{/operation}}
}
{{/operations}}
```
to
```
package {{package}};

{{#imports}}import {{import}};
{{/imports}}

import javax.ws.rs.*;
import javax.ws.rs.core.Response;

@Path("{{basePathWithoutHost}}")
public interface {{classname}}  {
{{#operations}}
{{#operation}}
    @{{httpMethod}}
    {{#subresourceOperation}}@Path("{{path}}"){{/subresourceOperation}}
    {{#hasConsumes}}@Consumes({ {{#consumes}}"{{mediaType}}"{{#hasMore}}, {{/hasMore}}{{/consumes}} }){{/hasConsumes}}
    {{#hasProduces}}@Produces({ {{#produces}}"{{mediaType}}"{{#hasMore}}, {{/hasMore}}{{/produces}} }){{/hasProduces}}
    public Response {{nickname}}({{#allParams}}{{>queryParams}}{{>pathParams}}{{>headerParams}}{{>bodyParams}}{{>formParams}}{{#hasMore}},
    {{/hasMore}}{{/allParams}});
{{/operation}}
}
{{/operations}}

```
(change is the @path annotation on the interface)

I just tested it now and it's ok for me :)